### ISSUE TITLE
question about the margin of loss? 

### ISSUE THREAD

**Issue by yuxiaokang-source** [KevinMusgrave/pytorch-metric-learning#277]
for example,If I have three class(class1,class 2,class3),class 1 and class 3(class 2 and class 3) is less similar than class1 and class 2, what should I do to reflect the information of similarity between class in loss function? Could I set different margins in a costom loss function?In my opinion, for example,the margin of TripletMarginLoss assumed all the class have equal similarity with each other.

**Comment by KevinMusgrave** [comment]
That's a good point. You could try MarginLoss with a learnable margin for each class:

```python
import torch
from pytorch_metric_learning import MarginLoss
loss_fn = MarginLoss(learn_beta = True, num_classes = num_classes)
loss_fn_optimizer = torch.optim.SGD(loss_fn.parameters(), lr=0.01)

for data in dataloader:
    ...
    loss_fn_optimizer.zero_grad()
    loss = loss_fn(embeddings, labels)
    loss.backward()
    loss_fn_optimizer.step()
```

If you want to write a custom loss with a margin per label, you can do something like this in the ```compute_loss``` function
```python
def compute_loss(self, embeddings, labels, indices_tuple):
    for L in labels:
        # You would set this array in __init__
        curr_margin = self.margins[L]
        curr_embeddings = embeddings[labels == L]
        ...
```
